app/src/main/java/in/testpress/testpress/authenticator/LoginActivity.java

```diff
@@ -200,6 +199,25 @@ public boolean onEditorAction(final TextView v, final int actionId,
         });
 
         usernameText.addTextChangedListener(watcher);
+        usernameText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (usernameText.getText().toString().contains(" ")) {
+                    usernameText.setText(usernameText.getText().toString().replaceAll(" ", ""));
```

Comment:
Only trailing space should be removed.

Reply:
Username/Email should not have spaces at all right?